lib/extract/dl_esm_inf/netcdf/kernel_data_netcdf.f90

```diff
@@ -72,29 +72,32 @@ module psy_data_mod
         integer                            :: next_var_index
     contains
         ! The various procedures used
-        procedure :: DeclareScalarInt,   WriteScalarInt,   ReadScalarInt
-        procedure :: DeclareScalarReal,  WriteScalarReal,  ReadScalarReal
-        procedure :: DeclareFieldDouble, WriteFieldDouble, ReadFieldDouble
-        procedure :: WriteScalarDouble
+        procedure :: DeclareScalarInt,    WriteScalarInt,    ReadScalarInt
+        procedure :: DeclareScalarReal,   WriteScalarReal,   ReadScalarReal
+        procedure :: DeclareFieldDouble,  WriteFieldDouble,  ReadFieldDouble
+        procedure :: DeclareScalarDouble, WriteScalarDouble, ReadScalarDouble
         procedure :: PreStart, PreEndDeclaration, PreEnd
         procedure :: PostStart, PostEnd
         procedure :: OpenRead
 
         !> The generic interface for declaring a variable:
-        generic, public :: PreDeclareVariable => DeclareScalarInt,  &
-                                                 DeclareScalarReal, &
+        generic, public :: PreDeclareVariable => DeclareScalarInt,    &
+                                                 DeclareScalarReal,   &
+                                                 DeclareScalarDouble, &
```

Comment:
Yes, it does work, but we need to compile in order to verify this. I've added this to #706 - with a dummy library this can be tested. I could also change a declaration in example/gocean/eg6 - should I do that as well?

Reply:
Yes please.

Reply:
Done.